core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -604,6 +604,27 @@ public void testCreateNamespace() {
     // Test with no metadata
     catalog.createNamespace(testNamespace);
     assertThat(catalog.namespaceExists(testNamespace)).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb", "ns1"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb", "ns1", "ns2"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testD_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb."))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb.ns"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb.ns_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb%"))).isFalse();
```

Comment:
For these false cases, can you run them twice? Once like this and once with a namespace that would match if the special character were not escaped.